AndroidPubSub/README.md

```diff
@@ -149,3 +149,72 @@ The keytool command does not allow importing an existing private key into a keys
         adb push <keystore name>.bks /data/user/0/your_app_dir_goes_here/files/<keystore name>
 
     The directory and filename used will depend on your use case.  Typically the application's files directory is in /data/user/0/<app namespace>/files/.  You may however choose to locate your keystore on removable media or another space on the filesystem.  The SDK allows for specifying the file path and name of the keystore so the choice is up to you.
+	
+## Using AWS IoT with authenticated users
+
+This sample demonstrates enabling unauthenticated users to use the AWS IoT APIs to securely publish-to and subscribe-from MQTT topics. Following steps demonstrates usage of AWS IoT with authenticated users. 
+
+1. Follow the steps [here](https://docs.aws.amazon.com/aws-mobile/latest/developerguide/add-aws-mobile-user-sign-in.html) to add authentication UI to your app.
+
+2. Get the credentials provider
+
+```
+credentialsProvider =
+               IdentityManager.getDefaultIdentityManager().getCredentialsProvider();
+```
+
+3. In the app, after the user logs in using the sign-in UI, get CognitoIdentity from credentialsProvider and use it to attach pricipal policy to the IotAndroidClient. Create IoT policy as specified in step 4 of [Using the sample](https://github.com/awslabs/aws-sdk-android-samples/tree/master/AndroidPubSub#using-the-sample) section above. 
+
+```
+// Setup the logins map
+IdentityManager identityManager = IdentityManager.getDefaultIdentityManager();
+Map<String, String> logins = new HashMap();
+IdentityProvider currentIdentityProvider = identityManager.getCurrentIdentityProvider();
+String token = currentIdentityProvider.getToken();
+String providerKey = currentIdentityProvider.getCognitoLoginKey();
+logins.put(providerKey, token);
+
+AmazonCognitoIdentity cognitoIdentity;
+GetIdRequest getIdReq;
+GetIdResult getIdRes;
+cognitoIdentity = new AmazonCognitoIdentityClient(credentialsProvider);
+getIdReq = new GetIdRequest();
+getIdReq.setLogins(logins);
+getIdReq.setIdentityPoolId(COGNITO_POOL_ID); //Identity pool Id generated by amplify cli
+new Thread(new Runnable() {
+    @Override
+    public void run() {
+	getIdRes = cognitoIdentity.getId(getIdReq);
+	AttachPrincipalPolicyRequest attachPolicyReq = new AttachPrincipalPolicyRequest();
+	attachPolicyReq.setPolicyName(AWS_IOT_POLICY_NAME);
+	attachPolicyReq.setPrincipal(getIdRes.getIdentityId());
+	AWSIotClient mIotAndroidClient = new AWSIotClient(credentialsProvider);
+	mIotAndroidClient.setRegion(Region.getRegion(MY_REGION));
+	mIotAndroidClient.attachPrincipalPolicy(attachPolicyReq);
+    }
+}).start();
+```
+
+3. Create MQTT client and connect
+
+```
+// MQTT Client
+// CUSTOMER_SPECIFIC_ENDPOINT is same as the one used in the sample app above
+mqttManager = new AWSIotMqttManager(clientId, CUSTOMER_SPECIFIC_ENDPOINT); 
+// Set keepalive to 10 seconds.  Will recognize disconnects more quickly but will also send
+// MQTT pings every 10 seconds.
+mqttManager.setKeepAlive(10);
```

Comment:
We are referring to a keep alive idea here, without introducing it in the README. Best to go with the default and not explicitly set it here.

Reply:
Fixed